ntoskrnl/include/internal/probe.h

```diff
@@ -31,7 +31,12 @@ DefaultSetInfoBufferCheck(ULONG Class,
 
         if (NT_SUCCESS(Status))
         {
-            if (PreviousMode != KernelMode)
+            /*
+             * An alignment of 1 (CHAR) means no alignment probe required.
+             * Some classes do not need alignment probing.
+             */
+            if (ClassList[Class].AlignmentSET != sizeof(CHAR) ||
+                PreviousMode != KernelMode)
```

Comment:
This is wrong. ProbeForRead fails for any address which is above highest user mode address, so any kernel mode call will fail in case Alignement != 1